src/setup/rgw.py

```diff
@@ -14,77 +14,228 @@
 # For any questions about this software or licensing,
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 
+import os
+import time
+import uuid
+import errno
+import shutil
+import socket
+from urllib.parse import urlparse
+from cortx.utils.validator.v_pkg import PkgV
+from cortx.utils.conf_store import Conf, MappedConf
+from cortx.utils.conf_store.error import ConfError
+from cortx.utils.process import SimpleProcess
+from cortx.utils.log import Log
+from src.setup.error import SetupError
+from src.const import (
+    CORTX_RPMS, CEPH_RPMS, CEPH_CONF_TMPL, CEPH_CONF, CEPH_ADMIN_KEYRING)
+
 
 class Rgw:
     """Represents RGW and Performs setup related actions."""
 
+    _machine_id = Conf.machine_id
+
     @staticmethod
     def validate(phase: str):
-        """Perform validtions."""
+        """Perform validations."""
+
+        Log.info(f'validations started for {phase} phase.')
+
+        if phase == 'post_install':
+            # Perform RPM validations
+            for rpms in [CORTX_RPMS, CEPH_RPMS]:
+                PkgV().validate('rpms', rpms)
+            Log.info(f'All RGW required RPMs are installed on {Rgw._machine_id} node.')
+        elif phase == 'prepare':
+            Rgw._file_exist(CEPH_CONF_TMPL)
+        elif phase == 'config':
+            Rgw._file_exist(CEPH_CONF)
+
+        Log.info(f'validations completed for {phase} phase.')
 
-        # Perform RPM validations
         return 0
 
     @staticmethod
-    def post_install(config_path: str):
-        """
-        Performs post install operations.
-        """
+    def post_install(conf: MappedConf):
+        """Performs post install operations."""
 
+        Log.info('PostInstall phase completed.')
         return 0
 
     @staticmethod
-    def prepare(config_path: str):
-        """
-        Performs prepare phase operations.
-        """
+    def prepare(conf: MappedConf):
+        """Prepare for operations required before RGW can be configured."""
+
+        Log.info('Prepare phase started.')
+
+        try:
+            hostname = conf.get(f'node>{Rgw._machine_id}>hostname')
+            host_ip = socket.gethostbyname(hostname)
+            ceph_tmpl_idx = 'ceph_conf_tmpl'
+            ceph_tmpl_url = f'ini://{CEPH_CONF_TMPL}'
+            # Load template ceph.conf file and update required field.
+            kv_list = [
+                ('global>fsid', str(uuid.uuid1())),
```

Comment:
As raised by Ujjwal, we need to confirm on need of FSID and 'mon host', where one monitor would be running per host, which doesn't seem to be right.
Keep a TODO

Reply:
fsid is not required.